src/Rings/mpolyquo-localizations.jl

```diff
@@ -182,13 +186,21 @@ modulus(R::MPAnyNonQuoRing)=ideal(R,[zero(R)])
 For ``L = (𝕜[x₁,…,xₙ]/I)[S⁻¹]`` this returns ``𝕜[x₁,…,xₙ]/I``.
 """
 quotient_ring(L::MPolyQuoLocalizedRing) = L.Q
+## 3 more signatures for compatibility to make quotient_ring agnostic
+quotient_ring(L::MPolyQuo) = L
+quotient_ring(L::MPolyRing) = L
+quotient_ring(L::MPolyLocalizedRing) = base_ring(L)
```

Comment:
Are you sure this is a good idea? This is not returning a quotient ring so commands like `modulus` are not applicable to the result. I don't have a clear idea on how this will be used, but maybe you want to return `quo(L, ideal(L, zero(L)))[1]` instead?

Reply:
modulus is already agnostic and needs to be (and I expect to add this for further commands when the need arises), because we cannot predict what kind of MPolyAnyRing a user might feed to it in their scripts. 
I was trying to stick to already existing rings as much as possible to not run into trouble with checks, whether objects live in the same ring, further down the road. But I am open for discussion on this subject, as long as we end up with quotient_ring and localized_ring being agnostic.

Reply:
I see the necessity for agnostic commands. But I think we should make sure that the output has the anticipated type in either case. For `modulus` this output type was mathematically motivated and I'm OK with that. But when I ask for a `quotient_ring` and get something else, this feels wrong. 

Yet, it might be a problem to organize caching of these. Do polynomial rings have attributes?

In general, I think, there's no way of avoiding input checks. If we want to support 4 different types of rings, then we most likely need to implement 4 different types of methods for every function we want to have.

Reply:
> I see the necessity for agnostic commands. But I think we should make sure that the output has the anticipated type in either case. For `modulus` this output type was mathematically motivated and I'm OK with that. But when I ask for a `quotient_ring` and get something else, this feels wrong.

I feel exactly the same way about this, but I did not find a good way to avoid creating new rings over and over again.
 
> Yet, it might be a problem to organize caching of these. Do polynomial rings have attributes?
We should discuss next week in Kaiserslautern, what kind of problems might arise with either solution. This seems like a crucial point not juat for this PR. 

> In general, I think, there's no way of avoiding input checks. If we want to support 4 different types of rings, then we most likely need to implement 4 different types of methods for every function we want to have.

Again I fully agree with you that up to a certain level, things have to be done 4 (or in some cases 4^2) times. On the other hand, at user level, this should not be visible and, to that end, we need to define the abstraction level at which this becomes agnostic and in what way to best implement this.

Reply:
I just tried it out: Both polynomial and quotient rings seem to allow attributes. Hence I would rewrite these methods as 
```julia
@attr MPolyQuo function quotient_ring(P::MPolyRing)
  return quo(P, ideal(P, zero(P)))[1]
end
```
Maybe this still throws in some cases since there are many concrete instances of `MPolyRing`. But if it does, we should just ask those responsible, whether we can make the rings in question attributable.